File: Examples/VisualBasic/WorkingWithCalendarExceptions/RetrieveCalendarExceptions.vb

Namespace WorkingWithCalendarExceptions
    Public Class RetrieveCalendarExceptions
        Public Shared Sub Run()
            ' ExStart:RetrieveCalendarExceptions
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' Create project instance
            Dim project1 As New Project(dataDir + "project_RetrieveExceptions_test.mpp")

            ' Access calendars
            For Each cal As Calendar In project1.Calendars
                ' Access calendar exceptions
                For Each calExc As CalendarException In cal.Exceptions
                    Console.WriteLine("From: " + calExc.FromDate.ToShortDateString())
                    Console.WriteLine("To: " + calExc.ToDate.ToShortDateString())
                Next
            Next
            ' ExEnd:RetrieveCalendarExceptions
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithCalendarExceptions/RetrieveCalendarExceptions.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithCalendarExceptions
{
    using System;

    internal class RetrieveCalendarExceptions
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:RetrieveCalendarExceptions
            //ExFor: Calendar.Exceptions
            //ExSummary: Shows how to retrieve info about calendar exceptions.
            var project = new Project(dataDir + "project_RetrieveExceptions_test.mpp");

            // Iterate over calendars
            foreach (var cal in project.Calendars)
            {
                // Access calendar exceptions
                foreach (var calExc in cal.Exceptions)
                {
                    Console.WriteLine("From: " + calExc.FromDate.ToShortDateString());
                    Console.WriteLine("To: " + calExc.ToDate.ToShortDateString());
                }
            }
            //ExEnd:RetrieveCalendarExceptions
        }
    }
}
